src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/DataContractResolverTestServiceHost.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.ServiceModel;
+using System.ServiceModel.Activation;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using System.Xml;
+
+namespace WcfService
+{
+    public class DataContractResolverTestServiceHostFactory : ServiceHostFactory
+    {
+        protected override ServiceHost CreateServiceHost(Type serviceType, Uri[] baseAddresses)
+        {
+            DataContractResolverTestServiceHost serviceHost = new DataContractResolverTestServiceHost(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class DataContractResolverTestServiceHost : TestServiceHostBase<IWcfService>
+    {
+        protected override string Address { get { return "DataContractResolver"; } }
+
+        protected override Binding GetBinding()
+        {
+            return new BasicHttpBinding();
+        }
+
+        protected override void ApplyConfiguration()
+        {
+            base.ApplyConfiguration();
+            this.AddServiceEndpoint(typeof(IWcfService), new BasicHttpBinding(), "");
+            foreach (ServiceEndpoint endpoint in this.Description.Endpoints)
+            {
+                foreach (OperationDescription operation in endpoint.Contract.Operations)
+                {
+                    if (operation.Name == "GetAllEmployees" || operation.Name == "AddEmployee")
```

Comment:
I think it might be better to make a new service interface.  IWcfService is getting overloaded.  This special casing of operation names is a hint these really belong somewhere else.  It would be hard to diagnose failures if someone refactored IWcfService.